File: temporal-io/my-python/MY/0_simple_split/2_multiple_nodes/worker.py
"""
Worker Script - Multiple Nodes
Runs the Temporal worker that processes multiple-node workflow and activity tasks.
"""
import asyncio
from concurrent.futures import ThreadPoolExecutor

from temporalio.client import Client
from temporalio.worker import Worker

from workflow_definitions import (
    MultipleNodesWorkflow,
    save_result,
    transform_data,
    validate_input,
)

# Configuration
TEMPORAL_HOST = "localhost:7233"
TASK_QUEUE = "0-simple-multiple-nodes-task-queue"
MAX_CONCURRENT_ACTIVITIES = 10


async def main():
    """Start and run the worker."""
    # Connect to Temporal server
    client = await Client.connect(TEMPORAL_HOST)
    print(f"Connected to Temporal at {TEMPORAL_HOST}")

    # Create and run worker
    worker = Worker(
        client,
        task_queue=TASK_QUEUE,
        workflows=[MultipleNodesWorkflow],
        activities=[validate_input, transform_data, save_result],
        activity_executor=ThreadPoolExecutor(MAX_CONCURRENT_ACTIVITIES),
    )

    print(f"Worker started on task queue: {TASK_QUEUE}")
    print(f"Max concurrent activities: {MAX_CONCURRENT_ACTIVITIES}")
    print("Registered activities: validate_input, transform_data, save_result")
    print("Press Ctrl+C to stop the worker")

    # Run the worker
    await worker.run()


if __name__ == "__main__":
    asyncio.run(main())

File: temporal-io/my-typescript/MY/0_simple_split/2_multiple_nodes/worker.ts
/**
 * Worker Script - Multiple Nodes
 * Runs the Temporal worker that processes multiple-node workflow and activity tasks.
 */
import { NativeConnection, Worker } from '@temporalio/worker';
import * as activities from './activities.js';
import { fileURLToPath } from 'url';
import path from 'path';

// Configuration
const TEMPORAL_HOST = 'localhost:7233';
const TASK_QUEUE = '0-simple-multiple-nodes-task-queue';

async function main() {
  try {
    // Connect to Temporal server
    const connection = await NativeConnection.connect({
      address: TEMPORAL_HOST,
    });
    console.log(`Connected to Temporal at ${TEMPORAL_HOST}`);

    // Create and run worker
    const __filename = fileURLToPath(import.meta.url);
    const __dirname = path.dirname(__filename);

    const worker = await Worker.create({
      connection,
      namespace: 'default',
      taskQueue: TASK_QUEUE,
      workflowsPath: path.join(__dirname, 'workflow-definitions.js'),
      activities,
    });

    console.log(`Worker started on task queue: ${TASK_QUEUE}`);
    console.log('Registered activities: validateInput, transformData, saveResult');
    console.log('Press Ctrl+C to stop the worker');

    // Run the worker
    await worker.run();
  } catch (error) {
    console.error('Worker error:', error);
    process.exit(1);
  }
}

main();
